### PR TITLE
[service.projcontrol] 1.2.0

### PR DESCRIPTION
**Pull request by fredrik-eriksson**
### Description

An addon for controling projectors connected via serial port. A more detailed description can be found on the project github page: https://github.com/fredrik-eriksson/kodi_projcontrol

It requires the pyserial module to work, and I've submitted that module in a separate PR. This addon can also optionally use flask to create a REST service to control the projector, however flask is not in the kodi repositories, and I have no will to submit and maintain it. As this addon is still usable (and hopefully even useful) even without the REST service I hope it's OK to include it anyway.

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: service.projcontrol/addon.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+<addon id="service.projcontrol"
+    name="Projector Control"
+    version="1.2.0"
+    provider-name="Fredrik Eriksson">
+  <requires>
+    <import addon="xbmc.python" version="2.19.0"/>
+    <import addon="script.module.pyserial" version="3.4.0"/>
+  </requires>
+  <extension point="xbmc.service" library="service.py" start="login" />
+  <extension point="xbmc.addon.metadata">
```
[the remark this answers is not in the document]
Comment (enen92) [reply service.projcontrol/addon.xml]: setting the `assets` element would be cool too. Note that the icon cannot have transparency and the fanart (while not mandatory) is advised. It is shown in our addon website and also in the addon browser.

**Comment by fredrik-eriksson** [comment]
Nice, thanks for the feedback. I had hoped to get this included by making the flask requirement optional, but I can understand why that won't work :)
I've never used bottle, but I'll try to port the API to it instead, thanks for the tip. I'm not really ready to drop the feature as I've found it quite useful.

Regarding platforms I figured this would work on any platform where python (and the dependencies) worked, as long as you know what device name to configure in pyserial. As far as I know I do not use anything that is platform specific. That said, I've only a Linux-machine to test with I'll guess I'll have to restrict it to Linux. The people who connect projectors by serial cables are probably Linux-users anyway :)

As Travis gave a warning about complex entry point I've also done some restructuring. Not sure if it became better or not, but I'll probably need some time to test and make sure everything still works before updating this pull request.

**Comment by enen92** [comment]
Its fine to keep it for `all` at the moment. If you find out it won't work on some platform you update it later

File: service.projcontrol/addon.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <platform>all</platform>
+    <summary lang="en">Control your projector from Kodi</summary>
+    <summary lang="sv">Hantera din projektor med Kodi</summary>
```
[the remark this answers is not in the document]
Comment (Rechi) [reply service.projcontrol/addon.xml]: This doesn't restrict anything. If <language>_<country> doesn't match your Kodi setting it then tries to match only <language> and then if that doesn't match too, it uses `en_GB` or `en`.

**Comment by petterreinholdtsen** [comment]
[Rechi]
> This doesn't restrict anything. If <language>_<country> doesn't match
> your Kodi setting it then tries to match only <language> and then if
> that doesn't match too, it uses `en_GB` or `en`.

Right, so Kodi is not working as described in RFC 3066, where
translations tagged with 'se_NO' would be ignored on systems with locale
set to se_SE?  Or translations tagged with pt_BR would be ignored on
systems with the locale set to pt_PT?

For systems behaving according to RFC 3066, I would tag translations
only with the 'se' language code to make sure systems with locale set to
both se_NO and se_SE would use the translation.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by fredrik-eriksson** [comment]
Sorry for the time taken, I think all requested changes has been honored now, and the bottle module seems to work just fine for this plugin. Travis seems happy except for the (obviously) missing pyserial plugin, so I hope this is good enough :)

**Comment by enen92** [comment]
Glad to know bottle worked fine for you. Unfortunately I can't merge with travis failing in this repository, we'll have to wait for pyserial to hit our repos (once the repository generator runs again) or someone with higher privileges than myself need to handle it. Pinging @Rechi

**Pull request merged**